relentless/optimize/optimizer.py

```diff
@@ -0,0 +1,23 @@
+__all__ = ['Optimizer','SteepestDescent']
+
+import abc
+
+from .objective import ObjectiveFunction
+
+class Optimizer(abc.ABC):
+    def __init__(self, obj):
+        self.objective = obj
+
+    @abc.abstractmethod
+    def run(self):
```

Comment:
Maybe `optimize` or `compute` instead?

I think we should change this interface slightly too. The Optimizer doesn't necessarily need to know the objective function when it is constructed, since it only needs it at the time it is called. (Or equivalently, you could use an Optimizer to minimize multiple objective functions using the same hyperparameters.)

So, I would move the objective function to be an argument of `run`. The constructor should be used for setting hyperparameters (like alpha below).